src/docs/sdk/client-reports.mdx

```diff
@@ -98,14 +98,23 @@ need to match the reason codes that relay would emit directly to Sentry.
 
 ## SDK Side Recommendations
 
-SDKs are encouraged to reduce the total amount of needless communciation.  As
-such a recommended approach is to keep track of the counts for the discard
-reasons in the transport directly and to periodically flush them out as separate
-envelope item or attach it to an already scheduled envelope.  As some SDKs still
-send legacy events instead of envelopes for backwards compatibility with older
-Sentry servers it would be recommended in such cases to send it as separate envelope
-instead or attach it to pending session envelopes.
-
-There is no expectation that such bookkeeping can work transparently for custom
-transports which is why it's generally acceptable if client reports are optional
-for custom transports.
+The client reports feature doesn't expect 100 percent correct numbers, and it is
+acceptable for the SDKs to lose a small number of client reports. The expectation of
+this feature is to give the users an approximation of specific outcomes. Of course,
+the SDKs should ensure not dropping too many reports. It is not required, for example: 
+ - to persist the data when an application crashes.
+ - to move an envelope item with a client report to the next envelope when the cache for envelopes is full. 
+
+SDKs are encouraged to reduce the total amount of needless communication, especially
+on mobile. A recommended approach is to keep track of the discard events in the
+transport directly and attach it to an already scheduled envelope or periodically flush
+them out as separate envelope items.
+For mobile SDKs, the recommendation is to attach the client report to the next envelope
+enqueued for submission instead of periodically flushing them out. We accept the tradeoff
```

Comment:
This part looks a bit weird, we mention that SDKs are encouraged to reduce the total of communication, but then we say: `or periodically flush
them out as separate envelope items.`

Reply:
I didn't write that initially. I guess the intention of @mitsuhiko was that you shouldn't send an envelope item each time an event is dropped. Especially on the backend you could have several items dropped per second and doing an HTTP request for each of them would be crazy. I will try to rephrase that a bit.

Reply:
Is it better now @marandaneto?

Reply:
yes, thanks.